scmgr/app.go

```diff
@@ -0,0 +1,562 @@
+	for i, s := range c.Args() {
+		log.Print(i, s)
+	}
+	cfg := getConfigOrFail(c)
+	sb := cfg.Sbm.GetFuzzy(c.Args().First())
+	if sb == nil {
+		return errors.New("didn't find latest block - update first")
+	}
+	client := skipchain.NewClient()
+	guc, cerr := client.GetUpdateChain(sb.Roster, sb.Hash)
+	if cerr != nil {
+		return cerr
+	}
+	latest := guc.Update[len(guc.Update)-1]
+	log.Print("Reading file", c.Args().Get(1))
+	data, err := ioutil.ReadFile(c.Args().Get(1))
+	log.ErrFatal(err)
+	ssbr, cerr := client.StoreSkipBlock(latest, nil, &html{data})
+	if cerr != nil {
+		return errors.New("while storing block: " + cerr.Error())
+	}
+	cfg.Sbm.Store(ssbr.Latest)
+	log.ErrFatal(cfg.save(c))
+	log.Infof("Added new block %x to chain %x", ssbr.Latest.Hash, ssbr.Latest.GenesisID)
+	return nil
+}
+
+// Updates a block to the latest block
+func update(c *cli.Context) error {
+	log.Info("Updating block")
+	if c.NArg() < 1 {
+		return errors.New("please give block-id to update")
+	}
+	cfg := getConfigOrFail(c)
+
+	sb := cfg.Sbm.GetFuzzy(c.Args().First())
+	if sb == nil {
+		return errors.New("didn't find latest block in local store")
+	}
+	client := skipchain.NewClient()
+	guc, cerr := client.GetUpdateChain(sb.Roster, sb.Hash)
+	if cerr != nil {
+		return errors.New("while updating chain: " + cerr.Error())
+	}
+	if len(guc.Update) == 1 {
+		log.Info("No new block available")
+	} else {
+		for _, b := range guc.Update[1:] {
+			log.Infof("Adding new block %x to chain %x", b.Hash, b.GenesisID)
+			cfg.Sbm.Store(b)
+		}
+	}
+	latest := guc.Update[len(guc.Update)-1]
+	log.Infof("Latest block of %x is %x", latest.GenesisID, latest.Hash)
+	log.ErrFatal(cfg.save(c))
+	return nil
+}
+
+// lsKnown shows all known skipblocks
+func lsKnown(c *cli.Context) error {
+	cfg, err := loadConfig(c)
+	if err != nil {
+		return errors.New("couldn't read config: " + err.Error())
+	}
+	if cfg.Sbm.Length() == 0 {
+		log.Info("Didn't find any blocks yet")
+		return nil
+	}
+	genesis := sbl{}
+	for _, sb := range cfg.Sbm.SkipBlocks {
+		if sb.Index == 0 {
+			genesis = append(genesis, sb)
+		}
+	}
+	sort.Sort(genesis)
+	for _, g := range genesis {
+		short := !c.Bool("long")
+		log.Info(g.Sprint(short))
+		sub := sbli{}
+		for _, sb := range cfg.Sbm.SkipBlocks {
+			if sb.GenesisID.Equal(g.Hash) {
+				sub = append(sub, sb)
+			}
+		}
+		sort.Sort(sub)
+		for _, sb := range sub {
+			log.Info("  " + sb.Sprint(short))
+		}
+	}
+	return nil
+}
+
+// lsIndex writes one index-file for every known skipchain and an index.html
+// for all skiplchains.
+func lsIndex(c *cli.Context) error {
+	output := c.Args().First()
+	if len(output) == 0 {
+		return errors.New("Missing output path")
+	}
+
+	cleanHTMLFiles(output)
+
+	cfg, err := loadConfig(c)
+	if err != nil {
+		return errors.New("couldn't read config: " + err.Error())
+	}
+
+	// Get the list of genesis block
+	genesis := sbl{}
+	for _, sb := range cfg.Sbm.SkipBlocks {
+		if sb.Index == 0 {
+			genesis = append(genesis, sb)
+		}
+	}
```

Comment:
Seems like this block of code is present multiple times. Might be good to factor it in its own utility method maybe ? As you deem fit.